### ISSUE TITLE
O que deveria ser público ou privado?

### ISSUE THREAD

**Issue by grvcoelho** [pagarme/handbook#3]

**Comment by filipedeschamps** [comment]
A gente deveria perguntar para o Gitlab (ou verificar de alguma outra forma) se, desde o marco zero, eles eram públicos ou privados e porque de uma dessas duas decisões.

De qualquer forma, eu voto por deixar público.

**Comment by MarcoWorms** [comment]
Ao meu ver o repo pode ser sempre público se não contém informações sentitivas da empresa como dados sobre a base de clientes e/ou chaves privadas. Vamos levantar uma checklist de motivos que fariam um repo não poder ser público?

**Comment by grvcoelho** [comment]
Concordo com o @MarcoWorms, mas precisamos **definir claramente o que não pode ser público**. 

Lembra que "público por padrão" corre o risco de colocarmos algo sensível publicamente. Então precisamos dessa "blacklist"

**Comment by felquis** [comment]
Não acho que alguns repos como pagarme-core caberiam bem publico neste momento, porém o handbook publico é suave d+

Também tornar publica conversas do cotidiano na empresa não parece muito interessante, não estamos agregando algo para a comunidade, já o handbook em si agrega muito.

Vou estudar sobre assunto :pick:

**Comment by grvcoelho** [comment]
Para mim, coisas que **NÃO PODEM** ser públicas:

1. Informações de clientes (quaisquer que sejam, desde nomes até outros documentos)
2. Informações pessoais de funcionários (salários, feedbacks, documentos, etc)
3. Qualquer tipo de informação financeira (cartão de crédito, dados bancários, etc)
4. Dados da nossa empresa que não são abertos para o mercado (parcerias, estratégia, metas, faturamento, balanço patrimonial, etc)
5. Problemas de segurança

Vocês conseguem pensar em algo mais que não pode ser público?

**Comment by osmanpontes** [comment]
# Blacklist

- Recursos privados
  - Projetos que são core da empresa: API. Gateway...
- Projetos que se forem abertos perdem o sentido
  - Repositório de desafios para candidatos
- Comentários se referindo projetos privados em issues de projetos públicos
  - Muitos não terão problema, mas aqui tem uma potencial falha de segurança
  - Exemplo: "resolvendo esse problema do projeto PUBLICO_X nós iremos resolver o problema de segurança A no projeto PRIVADO_Y"

**Comment by grvcoelho** [comment]
@osmanpontes: gostei bastante da sua definição.

No segundo ponto você diz `Projetos que se forem abertos perdem o sentido`. Acho que parte da discussão é: o que faz perder o sentido?

**Comment by mccraveiro** [comment]
Na minha opniao não perde o sentido abrir repositórios de desafios. Faz parte do mindset de um candidato iterar sobre o que já existe e saberemos facilmente se ele simplesmente copiou o código de outros. Isso também nos forca a sempre melhorar o processo.

**Comment by osmanpontes** [comment]
Eu não pensei em todos os casos, só vi que temos que lembrar do repositório de desafios.
Ele não é um produto core propriedade privada da empresa, mas se for aberto não pode ser utilizado.
Nele temos comentários sobre o que se espera do candidato e isso não faz sentido que o candidato saiba.
É meio difícil generalizar isso.

Que tal:
- Projetos que se foram abertos tem o seu uso comprometido

É meio amplo e parecido com o que escrevi antes, mas a ideia é essa.
Se ser público estraga o uso que você quer pra ele, então deve ser privado.

**Comment by osmanpontes** [comment]
@mccraveiro sugiro abrir outra issue para esse questionamento. Acredito que isso seja outro tópico de discussão, pois foi apenas um exemplo e não o assunto principal. :+1: Mas a discussão é boa.

**Comment by derekstavis** [comment]
>(...) perguntar para o Gitlab (...) se, desde o marco zero, eles eram públicos ou privados (...).

Gitlab sempre foi publico desde o marco zero, pois comecaram como uma startup baseada em um produto de carater open source. No handbook deles, porém, tem essa secão aqui:

https://about.gitlab.com/handbook/general-guidelines/

>12. Most things are public unless there is a reason not to. Not public by default are:
> * financial and legal information
> * individual job applications / compensation / feedback
> * partnerships with other companies
> * customer information in issues: if an issue needs to contain any specific information about a customer, including but not limited to company name, employee names, number of users, the issue should be made confidential.

**Comment by felquis** [comment]
É incrível ser 101% transparente, e quero caminhar para algo do tipo no futuro, porém algo que valorizo muito é **não ter medo de errar** aprendi isso no Lean Startup, em D+0 um novo funcionário tem que fazer deploy, se algo der errado fomos nós que fizemos errado não foi ele quem errou, e já sabemos o que deve acontecer quando algo da errado.

Vejo isso sendo muito passível de falha ainda mais tratando-se de 1500 colaboradores, pois toda a DLP deveria usar o sistemas de issues o mais próximo possível dos desenvolvedores e toda DLP deveria ser feita por desenvolvedores, resolvendo problemas do jeito certo e não do jeito que da.

O Celso tem ótimos pontos sobre áreas que podem ou não usar um canal público, por exemplo comercial, estratégia, risco... imagina clientes criando tickets de suporte no canal errado e passando informações sensíveis sem querer querendo, porque a vida é um aprendizado não é mesmo? Seria frustrante um colaborador novo ficar com medo de criar uma issue pois ele não tem certeza se isso é público ou privado.

Portanto, hoje já temos inúmeros repositórios que fazem sentido pra caralho ser Open Source como o https://github.com/pagarme/teleport, uma peça super importante para nossa empresa e Open Source com colaboradores fora da Pagar.me

![image](https://cloud.githubusercontent.com/assets/736728/23017540/d887face-f421-11e6-8f14-67ce4f568880.png)

Ou outros vários projetos Open Source, como todos os SDKs, Documentação, e outras pessoas fundamentais como o teleport.

Então, o que proponho é se for um software que é usado em nosso sistema, e que é muito bem abstraído dele, ele é Open Source by design. Agora discussões genericas da empresa como um todo também precisam estar o mais próximo possível, da maneira mais segura e mais fácil.

Hoje mesmo estava fazendo um email relacionado a fazer breakevens e dev talks no auditório, e não fazia sentido enviar nem por e-mail, nem em um canal público.

Foi com essa intenção que criei um https://github.com/pagarme/forum para justamente podermos centralizar em um local privado todas as discussões pertinentes a empresa, onde qualquer um da DLP deveria ter acesso. Onde quem quiser pode criar issues e tirar dúvidas, confirmar informações ou até mesmo conversar sobre onde é o melhor local para acontecer os breakevens onde comporte mais pessoas, ou solicitar água potável, televisões para monitoramento, **criar discussões que agreguem valor para nós e principalmente aberto para todos da empresa**

**Faze 1**: Estamos nela, temos uma empresa privada, em um mercado com muitas restrições pois lidamos com informações muito sensíveis, onde uma hora ou outra um cliente empreendedor que está começando acaba divulgando até mesmo dados sensíveis, portanto na fundação disso tudo deveria ter a oportunidade de errar sem medo, claro que todo mundo quer acertar e isso que é mais legal pois quando erramos o fazemos com maestria pois somos muito bons.

**Faze 2**: Hoje já temos inúmeros repositórios públicos e isso deveria ser algo com estratégia, pois algo que fazemos pode sobrecarregar uma área, hoje mesmo se quer nossos repositórios Open Source damos a antenção que eles precisam, e isso merece uma atenção pois isso reflete muito bem o que acontece aqui dentro, se queremos ser realmente bons não tem como fingir ser, tem que ser o melhor e ponto. Temos várias pessoas que amam Open Source de coração, e não temos um squad Open Source ainda, o Worms é a pessoa que mais brilha neste quisito no que vejo, hoje o @MarcoWorms @LeoNero e @joaolucasl trabalham 101% do tempo em projetos Open Source ou que serão publicos quando estiverem mais maduros.

Portanto recomendo super primeiro colocar toda a empresa com acesso ao `Pagar.me Acadamy`, este team deve ter acesso apenas aos repositórios Open Source educativos e apenas 1 repo privado o pagarme/forum, pois o fórum é algo que aqui dentro jamais pode morrer, as discussões em forum são ricas e as pessoas pensam para responder, e precisamos de pessoas pensando e com argumentos embasados para criar uma empresa world class.

**Faze 3**: Um é pouco, dois é bom, três é demais. Para tornar uma empresa privada em algo 101% público, precisamos de estratégia, e mesmo assim dificilmente será 101% pois a Terra é um local hostil e estamos aqui para mudar isso, enquanto aqui dentro não ter se quer água potável, não estamos mudando o mundo, estamos compactuando com ele.

That all folks!

**Comment by MarcoWorms** [comment]
BUMP, e ai, como vai esse assunto? Vamos concluir algo a partir do que ja foi falado? :D

**Comment by derekstavis** [comment]
:star2: Sumonando o @kamushadenes na thread

**Comment by felquis** [comment]
O que o seu coração mandar! :heart:

**Comment by grvcoelho** [comment]
> O que o seu coração mandar! :heart: 

@felquis: eu acho que não é bem assim não tá? Existem informações sensíveis (especialmente de clientes) que **não devem** ser públicas.

Deveríamos ter um guideline disso para não fazer o que o coração mandar.

**Comment by felquis** [comment]
Hahahaha você está muito correto, já levantamos o assunto no fórum https://github.com/pagarme/forum/issues/88

**Comment by Evilhideki** [comment]
Acho bacana essa discussão rolando aqui, e concordo com os pontos do exemplo do gitlab que o @derekstavis trouxe. MAs vocês pensaram em envolver o juridico nessa discussão?

No fim das contas, mesmo código que será "público"deverá ter aval deles, dado que essas ações podem ocasionar casos de jurisprudência.
